src/test/java/org/apache/sysml/test/integration/functions/paramserv/SparkDataPartitionerTest.java

```diff
@@ -0,0 +1,186 @@
+import org.apache.spark.api.java.function.Function2;
+import org.apache.sysml.api.DMLScript;
+import org.apache.sysml.parser.Statement;
+import org.apache.sysml.runtime.controlprogram.context.SparkExecutionContext;
+import org.apache.sysml.runtime.controlprogram.paramserv.ParamservUtils;
+import org.apache.sysml.runtime.controlprogram.paramserv.spark.DataPartitionerSparkMapper;
+import org.apache.sysml.runtime.controlprogram.paramserv.spark.DataPartitionerSparkReducer;
+import org.apache.sysml.runtime.matrix.data.MatrixBlock;
+import org.apache.sysml.runtime.matrix.data.MatrixIndexes;
+import org.apache.sysml.runtime.util.DataConverter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import scala.Tuple2;
+
+public class SparkDataPartitionerTest {
+
+	@Test
+	public void testSparkDataPartitionerDC() {
+		DataPartitionerSparkMapper mapper = new DataPartitionerSparkMapper(Statement.PSScheme.DISJOINT_CONTIGUOUS, 3);
+		Map<Integer, Tuple2<MatrixBlock, MatrixBlock>> result = doPartitioning(mapper);
+
+		Assert.assertEquals(3, result.size());
+		result.values().forEach(v -> {
+			Assert.assertEquals(v._1.getNumRows(), v._2.getNumRows());
+			double[] features = v._1.getDenseBlockValues();
+			double[] labels = v._2.getDenseBlockValues();
+			for (int i = 0; i < v._1.getNumRows(); i++) {
+				for (int j = 0; j < v._1.getNumColumns(); j++) {
+					Assert.assertEquals((int)labels[i], (int)features[i * v._1.getNumColumns() + j] / v._1.getNumColumns());
+				}
+			}
+		});
+	}
+
+	@Test
+	public void testSparkDataPartitionerDR() {
+		DataPartitionerSparkMapper mapper = new DataPartitionerSparkMapper(Statement.PSScheme.DISJOINT_RANDOM, 4);
+		Map<Integer, Tuple2<MatrixBlock, MatrixBlock>> result = doPartitioning(mapper);
+
+		List<MatrixBlock> pfs = result.values().stream().map(r -> r._1).collect(Collectors.toList());
+		List<MatrixBlock> pls = result.values().stream().map(r -> r._2).collect(Collectors.toList());
+		assertPermutationDR(4000 * 2000, pfs);
+		assertPermutationDR(4000, pls);
+	}
+
+	@Test
+	public void testSparkDataPartitionerDRR() {
+		DataPartitionerSparkMapper mapper = new DataPartitionerSparkMapper(Statement.PSScheme.DISJOINT_ROUND_ROBIN, 4);
+		Map<Integer, Tuple2<MatrixBlock, MatrixBlock>> result = doPartitioning(mapper);
+
+		Assert.assertEquals(4, result.size());
+		result.forEach((key, value) -> {
+			int workerID = key;
+			MatrixBlock features = value._1;
+			MatrixBlock labels = value._2;
+			Assert.assertEquals(features.getNumRows(), labels.getNumRows());
+			double[] fd = features.getDenseBlockValues();
+			double[] ld = labels.getDenseBlockValues();
+			for (int i = 0; i < features.getNumRows(); i++) {
+				Assert.assertEquals(workerID, ((int) ld[i]) % result.size());
+				for (int j = 0; j < labels.getNumColumns(); j++) {
+					Assert.assertEquals((int) ld[i],
+							(int) fd[i * features.getNumColumns() + j] / features.getNumColumns());
+				}
+			}
+		});
+	}
+
+
+	@Test
+	public void testSparkDataPartitionerOR() {
```

Comment:
Please compare with local OR partitioner to check for consistent global ordering.